internal/service/rate.go

```diff
@@ -0,0 +1,56 @@
+package service
+
+import (
+	"errors"
+	clients "github.com/GenesisEducationKyiv/software-engineering-school-4-0-DimaL-cloud/internal/client"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-DimaL-cloud/internal/client/rate"
+	"github.com/avast/retry-go/v4"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	RetriesAmount = 3
+)
+
+type RateService struct {
+	client rate.Rate
+}
+
+func NewRateService(clients *clients.Client) *RateService {
+	r := &RateService{}
+	r.client = clients.NbuRate
+	r.client.SetNext(clients.PrivatBankRate).
+		SetNext(clients.FawazahmedRate)
+	return r
+}
+
```

Comment:
It would be great if client code had ability to set clients order, instead of hardcoding it in rate service constructor

Reply:
👍